stripe/src/main/java/com/stripe/android/ApiRequestExecutor.kt

```diff
@@ -2,8 +2,9 @@ package com.stripe.android
 
 import com.stripe.android.exception.APIConnectionException
 import com.stripe.android.exception.InvalidRequestException
+import java.net.UnknownHostException
 
 internal interface ApiRequestExecutor {
-    @Throws(APIConnectionException::class, InvalidRequestException::class)
+    @Throws(APIConnectionException::class, InvalidRequestException::class, UnknownHostException::class)
```

Comment:
Hello, should Stripe also return a Stripe Exception if the exception type is SocketException or SocketTimeoutException?

In our app we consider a connection error the following

```
    private fun Throwable.isConnectionError() =
        this is UnknownHostException || this is SocketException || this is SocketTimeoutException
```

Reply:
I'm actually catching any `IOException`, which the exceptions you mentioned are subclasses of.